sdk/nodejs/resource.ts

```diff
@@ -278,8 +278,13 @@ export abstract class CustomResource extends Resource {
      * @param props The arguments to use to populate the new resource.
      * @param opts A bag of options that control this resource's behavior.
      */
-    constructor(t: string, name: string, props?: Inputs, opts?: CustomResourceOptions) {
+    constructor(t: string, name: string, props?: Inputs, opts: CustomResourceOptions = {}) {
+        if ((<ComponentResourceOptions>opts).providers) {
+            throw new ResourceError("Do not supply 'providers' option to a CustomResource. Did you mean 'provider' instead?", opts.parent);
+        }
+
         super(t, name, true, props, opts);
+        this.__pulumiCustomResource = true;
```

Comment:
Why'd you move this out of an initializer?

Reply:
annoyance in TS.  if you have a member initializer, you are not allowed to have pre-super checks.